--- git-orig.rb	2015-02-13 11:31:33.000000000 +0100
+++ git.rb	2015-02-13 12:14:38.000000000 +0100
@@ -68,7 +68,13 @@
 
   # Returns the readable identifier for the given git changeset
   def self.format_changeset_identifier(changeset)
-    changeset.revision[0, 8]
+	str = changeset.revision[0, 8]
+	h = changeset.repository.extra_info
+	unless h.nil? || h['refs'].nil?
+		ref = h['refs'][changeset.revision.to_s]
+		str += ' [' + ref.join(', ') + ']' if ref.respond_to?('join')
+	end
+	str
   end
 
   def branches
@@ -79,6 +85,10 @@
     scm.tags
   end
 
+  def refs
+    scm.refs
+  end
+
   def default_branch
     scm.default_branch
   rescue Exception => e
@@ -130,6 +140,8 @@
   def fetch_changesets
     scm_brs = branches
     return if scm_brs.nil? || scm_brs.empty?
+    merge_extra_info('refs' => refs)
+    self.save
 
     h1 = extra_info || {}
     h  = h1.dup
